packages/toast/src/Toast.tsx

```diff
@@ -0,0 +1,335 @@
+  opacity: 0;
+  transition: all ${transitionDuration}ms ease-in-out;
+`;
+
+const toastVariantStyles: Record<ToastVariants, string> = {
+  [ToastVariants.success]: css`
+    background-color: ${uiColors.green.light3};
+  `,
+
+  [ToastVariants.progress]: css`
+    background-color: ${uiColors.white};
+    padding-bottom: 6px;
+  `,
+};
+
+const toastVisibleStateStyle = css`
+  transform: translate3d(0, 0, 0) scale(1);
+  opacity: 1;
+`;
+
+const toastTransitionStateStyles: Partial<Record<RTGStates, string>> = {
+  [RTGStates.Entered]: toastVisibleStateStyle,
+};
+
+const baseIconStyle = css`
+  flex-shrink: 0;
+  margin-right: ${spacing[3]};
+`;
+
+const iconVariantStyles: Record<ToastVariants, string> = {
+  [ToastVariants.success]: css`
+    color: ${uiColors.green.base};
+  `,
+
+  [ToastVariants.progress]: css`
+    color: ${uiColors.gray.dark2};
+  `,
+};
+
+const contentWrapperStyles = css`
+  display: flex;
+  align-items: center;
+  padding: ${spacing[3]};
+`;
+
+const contentWrapperVariantStyles: Record<ToastVariants, string> = {
+  [ToastVariants.success]: css`
+    border-radius: 4px;
+    border: 1px solid ${uiColors.green.light2};
+  `,
+
+  [ToastVariants.progress]: css`
+    border-radius: 4px 4px 0 0;
+    border: 1px solid ${uiColors.blue.light2};
+    border-bottom: 0;
+  `,
+};
+
+const bodyVariantStyles: Record<ToastVariants, string> = {
+  [ToastVariants.success]: css`
+    color: ${uiColors.green.dark2};
+  `,
+
+  [ToastVariants.progress]: css`
+    color: ${uiColors.gray.dark2};
+  `,
+};
+
+const dismissButtonStyle = css`
+  position: absolute;
+  top: ${spacing[2]};
+  right: ${spacing[2]};
+  transition: color 0.15s ease-in-out;
+`;
+
+const dismissButtonVariantStyle = {
+  [ToastVariants.success]: css`
+    color: ${uiColors.green.base};
+
+    &:hover {
+      color: ${uiColors.green.dark2};
+
+      &:before {
+        background-color: ${uiColors.green.light2};
+      }
+    }
+  `,
+
+  [ToastVariants.progress]: '',
+};
+
+interface ToastProps extends Omit<React.ComponentProps<'div'>, 'title'> {
+  /**
+   * Optional text shown in bold above the body text.
+   */
+  title?: React.ReactNode;
+
+  /**
+   * Required main text for the Toast.
+   */
+  body: React.ReactNode;
+
+  /**
+   * Optional className passed to the wrapping <div /> for the toast.
+   */
+  className?: string;
+
+  /**
+   * Required style variant to render the Toast as.
+   */
+  variant: ToastVariants;
+
+  /**
+   * Optional number between 0 and 1 that sets the progress bar's progress. Note that the progress bar is only rendered when the Toast variant is set to `'progress'`.
+   *
+   * **Default:** `1`
+   */
+  progress?: number;
+
+  /**
+   * Optional boolean that renders the Toast and makes it visible when true.
+   *
+   * **Default:** `false`
+   */
+  open?: boolean;
+
+  /**
+   * Optional click event handler that, when set, renders a close button that receives the passed handler.
+   */
+  close?: React.MouseEventHandler;
```

Comment:
nit: banners and modals use `onClose` for a similar prop. Maybe worth renaming for consistency?

Reply:
Ah playing with it and I see how this works. nvm

Reply:
I named it `close` because it:
1. Doesn't fire when the component would close. The Toast is a fully-controlled component right now.
2. IT also enables a UI feature. By naming it `close`, I felt that it's more clear that defining the prop will also show the close button.